build.sbt

```diff
@@ -386,21 +380,59 @@ lazy val typed = akkaModule("akka-typed")
   )
   .disablePlugins(MimaPlugin)
 
-lazy val typedTestkit = akkaModule("akka-typed-testkit")
-  .dependsOn(typed, testkit % "compile->compile;test->test")
-  .settings(AutomaticModuleName.settings("akka.typed.testkit"))
+lazy val persistenceTyped = akkaModule("akka-persistence-typed")
+  .dependsOn(
+    actorTyped,
+    persistence,
+    testkit % "test->test",
+    typedTestkit % "test->test",
+    actorTypedTests % "test->test"
+  )
+  .settings(AkkaBuild.mayChangeSettings)
+  .settings(AutomaticModuleName.settings("akka.persistence.typed"))
   .disablePlugins(MimaPlugin)
 
-lazy val typedTests = akkaModule("akka-typed-tests")
+lazy val clusterTyped = akkaModule("akka-cluster-typed")
   .dependsOn(
-    typed,
-    typedTestkit % "compile->compile;test->provided;test->test",
-    // the provided dependencies
-    persistence % "compile->compile;test->test",
-    cluster % "test->test",
+    actorTyped,
+    cluster,
     clusterTools,
+    distributedData,
+    persistence % "provided->test",
+    persistenceTyped % "provided->test",
+    testkit % "test->test",
+    typedTestkit % "test->test",
+    actorTypedTests % "test->test"
+  )
+  .settings(AkkaBuild.mayChangeSettings)
+  .settings(AutomaticModuleName.settings("akka.cluster.typed"))
+  .disablePlugins(MimaPlugin)
+
+lazy val clusterShardingTyped = akkaModule("akka-cluster-sharding-typed")
+  .dependsOn(
+    clusterTyped,
+    persistenceTyped,
     clusterSharding,
-    distributedData
+    testkit % "test->test",
+    typedTestkit % "test->test",
+    actorTypedTests % "test->test"
+  )
+  .settings(AkkaBuild.mayChangeSettings)
+  .settings(AutomaticModuleName.settings("akka.cluster.sharding.typed"))
+  // To be able to import ContainerFormats.proto
+  .settings(Protobuf.importPath := Some(baseDirectory.value / ".." / "akka-remote" / "src" / "main" / "protobuf" ))
+  .disablePlugins(MimaPlugin)
+
+
+lazy val typedTestkit = akkaModule("akka-testkit-typed")
+  .dependsOn(actorTyped, testkit % "compile->compile;test->test")
+  .settings(AutomaticModuleName.settings("akka.testkit.typed"))
+  .disablePlugins(MimaPlugin)
+
+lazy val actorTypedTests = akkaModule("akka-actor-typed-tests")
```

Comment:
something missing for saying that this should not be published to maven central?

Reply:
That is a couple of lines below, `enablePlugins(NoPublish)`